### PR TITLE
Add Google reCAPTCHA

### PR DESCRIPTION
**Pull request by glombers**
https://www.google.com/recaptcha/

### CODE REVIEW

File: register.php [hunk 1]
```diff
@@ -28,6 +28,24 @@
...
 		
+		if(isset($_POST['g-recaptcha-response'])){
+          $captcha=$_POST['g-recaptcha-response'];
+        }
+		
+		if(!$captcha){
+          echo '<h2>Please check the the captcha form.</h2>';
+          exit;
+        }
+        $secretKey = "Put your secret key here";
```
Comment (ninjalulz) [inline register.php]: Indent the code properly, and the secret key variable should be placed in config.php.

Comment (Raggaer) [inline register.php]: Maybe conver this to a re-usable function so we can add the captcha on more places

Comment (Znote) [inline register.php]: Well, if we can get it to work properly without cURL, then that is the way to go. We want to be as dependency free as possible.

**Comment by Znote** [comment]
What about the existing captcha system that we already got installed? Perhaps remove that, replace the lines for that system with this google recaptcha?

File: register.php (hunk 1 above)
Comment (Raggaer) [inline register.php]: Whats wrong with dependencies?

Composer can take care of all of this hassle

Comment (Znote) [inline register.php]: I feel its like including jQuery just to hide an element on the page.

**Comment by glombers** [comment]
I understand I suggested to post Reggaer
`Your captcha can be easily bypassed with captcha killer services or neural networks / deep learning. I suggest google recpatcha
Site looks good. gl`

I'm allergic to such things. This fault that they do not know how much and I want to change something.

Your sytem captych is good and will not change

**Comment by Raggaer** [comment]
Simple text or number based captcha are so 2001. Your recpatcha PR is good but as @Znote pointed out some changes are needed!

**Comment by glombers** [comment]
I understand I do not have as many skills as you will be more pleased as we will do it together if Stefan thinks with a good solution. Add to that my poor English I have to hone what do you know: P

**Comment by Znote** [comment]
I just removed the old captcha system, implemented this google recaptcha, followed @Raggaer advice and went with curl.
Also changed recaptcha on the helpdesk and recovery page with this.
https://github.com/Znote/ZnoteAAC/commit/984e62e0959f3845d643b16d15c49effd3e86152

So closing this pull request.

**Pull request closed without merging**